src/Components/Users/UserProfile.tsx

```diff
@@ -22,32 +33,251 @@ const useStyles = makeStyles((theme: Theme) =>
+                        form: formData,
+                    });
+                }
+                setIsLoading(false);
+            }
+        },
+        [dispatchAction, username]
+    );
+    useAbortableEffect(
+        (status: statusType) => {
+            fetchData(status);
+        },
+        [fetchData]
+    );
+
 
     const handleChange = (panel: string) =>
         (event: React.ChangeEvent<{}>, isExpanded: boolean) => {
-        setExpanded(isExpanded ? panel : false);
+            setExpanded(isExpanded ? panel : false);
+        };
+
+    const handleChangeInput = (e: any) => {
+        let form = { ...states.form };
+        form[e.target.name] = e.target.value;
+        dispatch({ type: "set_form", form });
+    };
+
+
+    const validateForm = () => {
+        let errors = { ...initError };
+        let invalidForm = false;
+        Object.keys(states.form).forEach((field, i) => {
+            switch (field) {
+                case "first_name":
+                case "last_name":
+                case "gender":
+                    if (!states.form[field]) {
+                        errors[field] = "Field is required";
+                        invalidForm = true;
+                    }
+                    return;
+                case "age":
+                    if (!states.form[field]) {
+                        errors[field] = "This field is required";
+                        invalidForm = true;
+                    }
+                    return;
+                case "phone_number":
+                    const phoneNumber = parsePhoneNumberFromString(states.form[field]);
+                    if (!states.form[field] || !phoneNumber?.isPossible()) {
+                        errors[field] = "Please enter valid phone number";
+                        invalidForm = true;
+                    }
+                    return;
+                case "email":
+                    if (states.form[field] && !validateEmailAddress(states.form[field])) {
+                        errors[field] = "Enter a valid email address";
+                        invalidForm = true;
+                    }
+                    return;
+                default:
+                    return;
+            }
+        });
+        dispatch({ type: "set_error", errors });
+        return !invalidForm;
+    };
+
+
+    const handleValueChange = (phoneNo: any, name: string) => {
+        if (phoneNo && parsePhoneNumberFromString(phoneNo)?.isPossible()) {
+        const form = { ...states.form };
+        form[name] = phoneNo;
+        dispatch({ type: "set_form", form });
+        }
+    };
+
+
+    const handleSubmit = async (e: any) => {
+        e.preventDefault();
+        const validForm = validateForm();
+        if (validForm) {
+            setIsLoading(true);
+            const data = {
+                username: username,
+                first_name: states.form.first_name,
+                last_name: states.form.last_name,
```

Comment:
Codacy found an issue: [Identifier 'last_name' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5815458)